packages/@lwc/synthetic-shadow/src/faux-shadow/lwc-host.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { defineProperty, isNull } from '@lwc/shared';
+import { getAttribute, setAttribute } from '../env/element';
+
+/*
+ * Using a simple custom attribute is faster based on perf results:
+ * Setting attribute: https://jsperf.com/attribute-vs-data-attribute/1
+ * matches(selector): https://jsperf.com/matches-attribute-vs-data-attribute/1
+ * querySelector(): https://jsperf.com/query-by-attribute-vs-data-attribute/1
+ **/
+const HostTokenAttributeName = 'lwc:host';
+
+/**
+ * Patching HTMLElement.prototype.$lwcHostToken$ to add a custom attribute to host elements
+ */
+defineProperty(HTMLElement.prototype, '$lwcHostToken$', {
+    get(this: HTMLElement): boolean {
+        return !isNull(getAttribute.call(this, HostTokenAttributeName));
+    },
+    set(this: HTMLElement, value: string) {
+        setAttribute.call(this, HostTokenAttributeName, value);
```

Comment:
should the value be boolean? if the value can be anything, then the getter should return the value not boolean... you see what I mean?

Reply:
Value should be a strong as per spec. Anything else provided will be coerced to a string.